File: app/src/main/java/jgappsandgames/smartreminderslite/home/AboutActivity.java
package jgappsandgames.smartreminderslite.home;

// Java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.net.MalformedURLException;
import java.net.URL;
import java.net.URLConnection;

// Android OS
import android.app.Activity;
import android.content.Intent;
import android.net.Uri;
import android.os.Bundle;

// Views
import android.widget.TextView;
import android.widget.Toast;

// JSON
import org.json.JSONException;
import org.json.JSONObject;

// App
import jgappsandgames.smartreminderslite.BuildConfig;
import jgappsandgames.smartreminderslite.R;

/**
 * AboutActivity
 * Created by joshua on 10/2/17.
 */
public class AboutActivity extends Activity {
    @Override
    public void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_about);

        // Set Text
        ((TextView) findViewById(R.id.version)).setText(BuildConfig.VERSION_NAME);
        ((TextView) findViewById(R.id.build)).setText(String.valueOf(BuildConfig.VERSION_CODE));
        ((TextView) findViewById(R.id.api)).setText(jgappsandgames.smartreminderssave.BuildConfig.VERSION_NAME);

        // Check For Updates
        findViewById(R.id.update).setOnClickListener((view -> {
            Thread thread = new Thread(() -> {
                try {
                    final URL release = new URL("https://www.dropbox.com/s/a0i0ieed2dpskoo/release.json?dl=1");
                    final URLConnection release_connection = release.openConnection();
                    final BufferedReader reader = new BufferedReader(new InputStreamReader(release_connection.getInputStream()));

                    StringBuilder s = new StringBuilder();
                    while (true) {
                        final String t = reader.readLine();
                        if (t == null) break;
                        else s.append(t);
                    }

                    JSONObject data = new JSONObject(s.toString());
                    if (data.optInt("stable", 0) > BuildConfig.VERSION_CODE) {
                        runOnUiThread(() -> {
                            Toast.makeText(this, "Update Found", Toast.LENGTH_LONG).show();
                            startActivity(new Intent(Intent.ACTION_VIEW, Uri.parse("https://github.com/tfcbandgeek/SmartReminders-android/releases")));
                        });
                    } else {
                        runOnUiThread(() -> Toast.makeText(this, "No Update Found", Toast.LENGTH_LONG).show());
                    }
                } catch (MalformedURLException e) {
                    e.printStackTrace();
                } catch (IOException i) {
                    i.printStackTrace();
                } catch (JSONException j) {
                    j.printStackTrace();
                } catch (NullPointerException n) {
                    n.printStackTrace();
                } catch (Exception e) {
                    e.printStackTrace();
                }
            });

            thread.start();
        }));

        // Release Candidate Updates
        findViewById(R.id.update_rc).setOnClickListener((view -> {
            Thread thread = new Thread(() -> {
                try {
                    final URL release = new URL("https://www.dropbox.com/s/a0i0ieed2dpskoo/release.json?dl=1");
                    final URLConnection release_connection = release.openConnection();
                    final BufferedReader reader = new BufferedReader(new InputStreamReader(release_connection.getInputStream()));

                    StringBuilder s = new StringBuilder();
                    while (true) {
                        final String t = reader.readLine();
                        if (t == null) break;
                        else s.append(t);
                    }

                    JSONObject data = new JSONObject(s.toString());
                    if (data.optInt("rc", 0) > BuildConfig.VERSION_CODE) {
                        runOnUiThread(() -> {
                            Toast.makeText(this, "Update Found", Toast.LENGTH_LONG).show();
                            startActivity(new Intent(Intent.ACTION_VIEW, Uri.parse("https://github.com/tfcbandgeek/SmartReminders-android/releases")));
                        });
                    } else {
                        runOnUiThread(() -> Toast.makeText(this, "No Update Found", Toast.LENGTH_LONG).show());
                    }
                } catch (MalformedURLException e) {
                    e.printStackTrace();
                } catch (IOException i) {
                    i.printStackTrace();
                } catch (JSONException j) {
                    j.printStackTrace();
                } catch (NullPointerException n) {
                    n.printStackTrace();
                } catch (Exception e) {
                    e.printStackTrace();
                }
            });

            thread.start();
        }));

        // Beta Updates
        findViewById(R.id.update_beta).setOnClickListener((view -> {
            Thread thread = new Thread(() -> {
                try {
                    final URL release = new URL("https://www.dropbox.com/s/a0i0ieed2dpskoo/release.json?dl=1");
                    final URLConnection release_connection = release.openConnection();
                    final BufferedReader reader = new BufferedReader(new InputStreamReader(release_connection.getInputStream()));

                    StringBuilder s = new StringBuilder();
                    while (true) {
                        final String t = reader.readLine();
                        if (t == null) break;
                        else s.append(t);
                    }

                    JSONObject data = new JSONObject(s.toString());
                    if (data.optInt("beta", 0) > BuildConfig.VERSION_CODE) {
                        runOnUiThread(() -> {
                            Toast.makeText(this, "Update Found", Toast.LENGTH_LONG).show();
                            startActivity(new Intent(Intent.ACTION_VIEW, Uri.parse("https://github.com/tfcbandgeek/SmartReminders-android/releases")));
                        });
                    } else {
                        runOnUiThread(() -> Toast.makeText(this, "No Update Found", Toast.LENGTH_LONG).show());
                    }
                } catch (MalformedURLException e) {
                    e.printStackTrace();
                } catch (IOException i) {
                    i.printStackTrace();
                } catch (JSONException j) {
                    j.printStackTrace();
                } catch (NullPointerException n) {
                    n.printStackTrace();
                } catch (Exception e) {
                    e.printStackTrace();
                }
            });

            thread.start();
        }));

        // Alpha Updates
        findViewById(R.id.update_alpha).setOnClickListener((view -> {
            Thread thread = new Thread(() -> {
                try {
                    final URL release = new URL("https://www.dropbox.com/s/a0i0ieed2dpskoo/release.json?dl=1");
                    final URLConnection release_connection = release.openConnection();
                    final BufferedReader reader = new BufferedReader(new InputStreamReader(release_connection.getInputStream()));

                    StringBuilder s = new StringBuilder();
                    while (true) {
                        final String t = reader.readLine();
                        if (t == null) break;
                        else s.append(t);
                    }

                    JSONObject data = new JSONObject(s.toString());
                    if (data.optInt("alpha", 0) > BuildConfig.VERSION_CODE) {
                        runOnUiThread(() -> {
                            Toast.makeText(this, "Update Found", Toast.LENGTH_LONG).show();
                            startActivity(new Intent(Intent.ACTION_VIEW, Uri.parse("https://github.com/tfcbandgeek/SmartReminders-android/releases")));
                        });
                    } else {
                        runOnUiThread(() -> Toast.makeText(this, "No Update Found", Toast.LENGTH_LONG).show());
                    }
                } catch (MalformedURLException e) {
                    e.printStackTrace();
                } catch (IOException i) {
                    i.printStackTrace();
                } catch (JSONException j) {
                    j.printStackTrace();
                } catch (NullPointerException n) {
                    n.printStackTrace();
                } catch (Exception e) {
                    e.printStackTrace();
                }
            });

            thread.start();
        }));
    }
}

File: app/src/main/java/jgappsandgames/smartreminderslite/home/AboutActivity.kt
package jgappsandgames.smartreminderslite.home

// Java
import java.io.BufferedReader
import java.io.InputStreamReader
import java.net.URL

// Android OS
import android.content.Intent
import android.net.Uri
import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity

// Anko
import org.jetbrains.anko.toast

// JSON
import org.json.JSONObject

// App
import jgappsandgames.smartreminderslite.BuildConfig
import jgappsandgames.smartreminderslite.R

// KotlinX
import kotlinx.android.synthetic.main.activity_about.*

/**
 * AboutActivity
 * Created by joshua on 12/30/2017.
 */
class AboutActivity: AppCompatActivity() {
    public override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_about)
        setTitle(R.string.app_name)

        // Set Text
        version.text = BuildConfig.VERSION_NAME
        build.text = BuildConfig.VERSION_CODE.toString()
        api.text = jgappsandgames.smartreminderssave.BuildConfig.VERSION_NAME

        // Check For Updates
        update.setOnClickListener {
            val thread = Thread {
                try {
                    val release = URL("https://www.dropbox.com/s/a0i0ieed2dpskoo/release.json?dl=1")
                    val releaseConnection = release.openConnection()
                    val reader = BufferedReader(InputStreamReader(releaseConnection.getInputStream()))

                    val s = StringBuilder()
                    while (true) {
                        val t = reader.readLine()
                        if (t == null) break
                        else s.append(t)
                    }

                    val data = JSONObject(s.toString())
                    if (data.optInt("stable", 0) > BuildConfig.VERSION_CODE) {
                        runOnUiThread {
                            startActivity(Intent(Intent.ACTION_VIEW, Uri.parse("https://github.com/tfcbandgeek/SmartReminders-android/releases")))
                        }
                    } else {
                        runOnUiThread { toast(R.string.no_update_found).show() }
                    }
                } catch (e: Exception) {
                    e.printStackTrace()
                }
            }

            thread.start()
        }

        // External Libraries
        anko_button.setOnClickListener {
            startActivity(Intent(Intent.ACTION_VIEW, Uri.parse(getString(R.string.anko_path))))
        }

        fab_button.setOnClickListener {
            startActivity(Intent(Intent.ACTION_VIEW, Uri.parse(getString(R.string.fab_path))))
        }

        kotlin_button.setOnClickListener {
            startActivity(Intent(Intent.ACTION_VIEW, Uri.parse(getString(R.string.kotlin_path))))
        }

        pool_button.setOnClickListener {
            startActivity(Intent(Intent.ACTION_VIEW, Uri.parse(getString(R.string.pool_utility_path))))
        }
    }
}